### PR TITLE
blob: add common-prefix entries to synchronous list

### PR DESCRIPTION
**Pull request by LihaoLiuXs**
## Summary

- add opt-in `ListBlobsRequest.withIncludeCommonPrefixes(true)` support for synchronous blob listing
- mark delimiter-derived entries through `BlobInfo.isCommonPrefix()` while preserving the default blob-only behavior
- implement provider behavior for AWS S3, GCP GCS, Alibaba OSS, and InMemory, including prefix-only pagination
- keep GCP managed/empty folders excluded while still returning delimiter-derived common prefixes
- represent an InMemory marker-object collision once, as a common-prefix entry
- add shared conformance coverage, provider-focused tests, and recorded AWS/GCP fixtures
- skip only the Alibaba replay case until its dedicated recording exists; record mode remains enabled

## Tracking

- [TD-0336774](https://gus.lightning.force.com/lightning/r/ADM_Team_Dependency__c/a0nEE000001MqdhYAC/view)
- [W-24013626](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00002iz8T9YAI/view)
- [W-24013642](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00002iyQ4YYAU/view)

## Validation

- provider unit suites: 1,002 tests, 0 failures/errors; 4 benchmark tests skipped by design
- focused AWS and GCP conformance tests: record mode and credential-free replay passed
- focused Alibaba conformance replay: conditionally skipped pending its dedicated recording
- full replay verification through all blob modules: AWS, Alibaba, GCP, and InMemory passed
- staged diff checks and credential-value scan passed

Alibaba conformance recording remains pending execution in the dedicated Alibaba environment.

### CODE REVIEW

**Comment by LihaoLiuXs** [comment]
Revision `5140f100` addresses the latest review findings:

1. Alibaba replay no longer fails CI before its dedicated recording exists. The conformance override runs in record mode and automatically enables replay once matching mappings are present; Alibaba provider behavior is covered by a focused unit test now.
2. GCP always requests `includeFolders(false)`, so managed and empty folders are not surfaced. Delimiter-derived prefixes are still returned and marked as common prefixes; the GCP recording was refreshed for that request.
3. InMemory marker collisions now produce one common-prefix entry instead of both a marker blob and a prefix. A regression test covers marker plus child-object input.
4. Provider-focused coverage now exists for GCP, Alibaba, and InMemory in addition to the shared/AWS coverage.

Validation: 1,002 provider unit tests passed with zero failures/errors; focused AWS and GCP record/replay conformance tests passed; the focused Alibaba replay test is skipped only until its recording is generated. The staged diff and recording changes were also scanned for credential values.

File: blob/blob-aws/src/main/java/com/salesforce/multicloudj/blob/aws/BlobInfoIterator.java [hunk 1]
```diff
@@ -69,10 +87,12 @@ public boolean hasNext() {
       return true;
     }
 
-    if (nextContinuationToken != null) {
+    while (nextContinuationToken != null) {
```
Comment (roseyang62) [inline blob/blob-aws/src/main/java/com/salesforce/multicloudj/blob/aws/BlobInfoIterator.java]: Good fix changing `if` → `while` — it stops an empty intermediate page (keys all rolled into prefixes) from ending iteration early and dropping later pages.

Could we add a regression test though? All current tests stub a single page with a null token, so the loop never runs — revert this to `if` and everything still passes while the bug returns. A two-page mock (`thenReturn(page1, page2)`: page1 empty + token, page2 has blobs) would cover it. The Ali iterator has the same change and gap.

File: blob/blob-gcp/src/main/java/com/salesforce/multicloudj/blob/gcp/GcpBlobStore.java [hunk 2]
```diff
@@ -544,7 +547,7 @@ protected Iterator<com.salesforce.multicloudj.blob.driver.BlobInfo> doList(
       // i.e., Subsequent page responses are only fetched when the iterator is advanced.
       private final Iterator<Blob> blobIterator = Iterators.filter(
           blobs.iterator(),
-          blob -> !blob.isDirectory()
+          blob -> includeCommonPrefixes || !blob.isDirectory()
```
Comment (roseyang62) [inline blob/blob-gcp/src/main/java/com/salesforce/multicloudj/blob/gcp/GcpBlobStore.java]: Consistency issue: AWS/Ali/InMemory sort the merged blobs + common prefixes by key, but this GCP path only filters and doesn't sort, so ordering differs.

E.g. `b.txt` + `a/x` with `delimiter="/"`: others return `["a/", "b.txt"]`, GCP returns `["b.txt", "a/"]` — same data, different order per provider. Could we sort each page with the same comparator? (Or drop the sorts elsewhere + document that order isn't guaranteed.) The conformance test uses a `HashSet`, so it doesn't catch this.
